ansible_events_ui/ruleset.py

```diff
@@ -56,57 +70,85 @@ async def activate_rulesets(
+            *cmd_args,
+            cwd=local_working_directory,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.STDOUT,
+        )
 
-    # initial version using docker
-    # mounting volumes is probably the wrong way to do this
-    # it would be better to build it into the container or pass it
-    # through a stream (stdin or socket)
-    # cmd = f"docker run -v {rules_file}:/rules.yml -v {inventory_file}:/inventory.yml -v {vars_file}:/vars.yml -it {execution_environment} ansible-events --rules /rules.yml -i /inventory.yml --vars /vars.yml"  # noqa
-
-    # try this with podman
-    # cmd = f"podman run -v {rules_file}:/rules.yml -v {inventory_file}:/inventory.yml -v {vars_file}:/vars.yml -it {execution_environment} ansible-events --rules /rules.yml -i /inventory.yml --vars /vars.yml"  # noqa
-
-    # for local development this is better
-    cmd = [
-        "--rules",
-        rules_file,
-        "-i",
-        inventory_file,
-        "--vars",
-        vars_file,
-        "--websocket-address",
-        "ws://localhost:8080/api/ws2",
-        "--id",
-        str(activation_id),
-    ]
-    logger.debug(ansible_events)
-    logger.debug(cmd)
-
-    proc = await asyncio.create_subprocess_exec(
-        ansible_events,
-        *cmd,
-        cwd=tempdir,
-        stdout=asyncio.subprocess.PIPE,
-        stderr=asyncio.subprocess.STDOUT,
-    )
+        activated_rulesets[activation_id] = proc
 
-    activated_rulesets[activation_id] = proc
+        task = asyncio.create_task(
+            read_output(proc, activation_id, db),
+            name=f"read_output {proc.pid}",
+        )
+        taskmanager.tasks.append(task)
+
+    elif (
+        settings.deployment_type == "docker"
+        or settings.deployment_type == "podman"
+    ):
+
+        docker = aiodocker.Docker()
+
+        if settings.deployment_type == "docker":
+            host = "host.docker.internal"
+            port = settings.port
+        else:
+            host = settings.server_name
+            port = settings.port
+
+        container = await docker.containers.create(
+            {
+                "Cmd": [
+                    "ssh-agent",
+                    "ansible-events",
+                    "--worker",
+                    "--websocket-address",
+                    f"ws://{host}:{port}/api/ws2",
+                    "--id",
+                    str(activation_id),
+                ],
+                "Image": execution_environment,
+                "Env": ["ANSIBLE_FORCE_COLOR=True"],
+                "ExtraHosts": ["host.docker.internal:host-gateway"],
+            }
+        )
+        await container.start()
+
+        activated_rulesets[activation_id] = container
 
-    return cmd, proc
+        task = asyncio.create_task(
+            read_log(container, activation_id, db),
+            name=f"read_log {container}",
+        )
+        taskmanager.tasks.append(task)
+
+    elif settings.deployment_type == "k8s":
```

Comment:
Please add some warning log message, that K8S deployment type is not supported at the moment?